detectable/src/main/java/com/synopsys/integration/detectable/detectables/gradle/inspection/parse/GradleReportLineParser.java

```diff
@@ -77,9 +92,53 @@ private List<String> parseGav(String line) {
             }
         }
 
+        String projectName = metadata.getOrDefault(PROJECT_NAME_PREFIX, "orphanProject");
+        String rootProjectName = metadata.getOrDefault(ROOT_PROJECT_NAME_PREFIX, "");
+        String projectParent = metadata.getOrDefault(PROJECT_PARENT_PREFIX, null);
+
+        addRelation(projectParent, projectName, rootProjectName);
+
+        if(gavPieces.size() == 3) {
+            String dependencyGroupName = gavPieces.get(0) + ":" + gavPieces.get(1);
+            if ((cleanedOutput.contains(STRICTLY) || cleanedOutput.contains(REJECT) || cleanedOutput.contains(REQUIRE) || cleanedOutput.contains(PREFER))) {
+                gradleRichVersions.putIfAbsent(dependencyGroupName, gavPieces.get(2));
+                richVersionGroup.computeIfAbsent(projectName, value -> new HashSet<>()).add(dependencyGroupName);
+            }
+
+            if(gradleRichVersions.containsKey(dependencyGroupName)) {
+                if(richVersionGroup.containsKey(projectName) && richVersionGroup.get(projectName).contains(dependencyGroupName)) {
+                    gavPieces.set(2, gradleRichVersions.get(dependencyGroupName));
+                } else if(richVersionGroup.containsKey(rootProjectName) && richVersionGroup.get(rootProjectName).contains(dependencyGroupName)) {
+                    gavPieces.set(2, gradleRichVersions.get(dependencyGroupName));
+                } else if (checkParentRichVersion(rootProjectName, projectParent, dependencyGroupName)) {
+                    gavPieces.set(2, gradleRichVersions.get(dependencyGroupName));
+                }
+            }
+        }
+
         return gavPieces;
     }
 
+    private void addRelation(String projectParent, String projectName, String rootProjectName) {
+        if (!projectParent.equals("null") && !projectParent.contains("root project")) {
+            String parentString = projectParent.substring(projectParent.lastIndexOf(":") + 1, projectParent.lastIndexOf("'"));
+            relationsMap.putIfAbsent(projectName, parentString);
+        } else if (!projectParent.equals("null") && !projectName.equals(rootProjectName)) {
+            relationsMap.putIfAbsent(projectName, rootProjectName);
+        }
```

Comment:
We get the immediate parent information in the metadata from the generated gradle dependency graph file during runtime. This would ensure a Map is filled up with project as the key and their parents as the value. Examples of what the project parents would like are `project ':sqldelight-compiler'`, `root project 'sqldelight'`. This has been tested for many projects and this is the gradle way for describing a module according to the documentation.